### PR TITLE
Add comprehensive tests for FileManagerImpl

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

Test coverage improvements for critical file management functionality.

## 📔 Objective

This PR adds comprehensive test coverage for `FileManagerImpl`, the component responsible for file I/O operations in the Authenticator application. File handling is a critical operation for import/export functionality and requires rigorous testing to ensure data integrity.

**Test Coverage:**
- File writing operations with various data types and encodings
- File reading with error handling and edge cases
- Directory creation and management
- File deletion and cleanup operations
- Permission handling and error propagation
- Content URI handling for Android storage framework

The FileManager component handles sensitive user data during import/export operations. These tests validate correct file operations, proper error handling, and secure data handling across various Android storage scenarios.

## 📸 Screenshots

N/A - Test-only changes

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        val mockOutputStream = createMockOutputStream(capturedBytes)
+
+        every { mockContentResolver.openOutputStream(mockUri) } returns mockOutputStream
+
+        val result = fileManager.stringToUri(mockUri, testString)
+
+        assertTrue(result)
+        assertEquals(testString.length, capturedBytes.size)
+        assertEquals(testString, String(capturedBytes.toByteArray()))
+    }
```
[the remark this answers is not in the document]
Comment (SaintPatrck) [reply data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt:97]: @claude Good catch! The erroneous behavior of `stringToUri` is corrected in https://github.com/bitwarden/android/pull/6432.

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+                position += toRead
+                toRead
+            }
+            every { close() } just runs
+        }
+
+        every { mockContentResolver.openInputStream(mockUri) } returns mockInputStream
+
+        val result = fileManager.uriToByteArray(mockUri)
+        val expected = Result.success(testData)
```
Comment (david-livefront) [inline data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt]: Can we use the `asSuccess()` extension

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        runTest {
+            every { mockContentResolver.openInputStream(mockUri) } returns null
+
+            val result = fileManager.uriToByteArray(mockUri)
+
+            assertTrue(result.isFailure)
+            val exception = result.exceptionOrNull()
+            assertNotNull(exception)
+            assertEquals(IllegalStateException::class.java, exception!!.javaClass)
+            assertEquals("Stream has crashed", exception.message)
```
Comment (david-livefront) [inline data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt:133]: Can we clean up some of these assertions

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+        val mockInputStream = createMockInputStream(testData)
+
+        every { mockContentResolver.openInputStream(mockUri) } returns mockInputStream
+
+        val result = fileManager.uriToByteArray(mockUri)
+        val expected = testData.asSuccess()
+
+        assertTrue(result.isSuccess)
+        assertTrue(expected.isSuccess)
+        assertArrayEquals(expected.getOrNull(), result.getOrNull())
```
Comment (david-livefront) [inline data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt]: Can we simplify this:

```kotlin
val result = fileManager.uriToByteArray(mockUri)

assertArrayEquals(testData, result.getOrThrow())
```

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+            every { close() } just runs
+        }
+
+        every { mockContentResolver.openInputStream(mockUri) } returns mockInputStream
+
+        val result = fileManager.uriToByteArray(mockUri)
+        val expected = testData.asSuccess()
+
+        assertTrue(result.isSuccess)
+        assertTrue(expected.isSuccess)
```
Comment (david-livefront) [inline data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt]: Can we ditch the `expected` value and just compare against the `testData`?

File: data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+        val testData = ByteArray(0)
+        val mockInputStream = createMockInputStream(testData)
+
+        every { mockContentResolver.openInputStream(mockUri) } returns mockInputStream
+
+        val result = fileManager.uriToByteArray(mockUri)
+        val expected = testData.asSuccess()
+
+        assertTrue(result.isSuccess)
+        assertTrue(expected.isSuccess)
```
Comment (david-livefront) [inline data/src/test/kotlin/com/bitwarden/data/manager/file/FileManagerTest.kt]: Ditto

**Review by david-livefront — APPROVED** [review]
